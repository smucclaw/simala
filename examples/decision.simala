#eval let
  
  get_opinion =

    fun (b, inv) =>

    if
            b.stage     == 'seed
         && b.sector    == 'information_technology
         && b.stage_com == 'pre_revenue
          
      ||    b.stage     == 'series_a
         && b.sector    == 'information_technology
         && b.stage_com == 'pre_profit
          
      || inv.has_ESG

    then 'interesting
    else 'reject

in

  get_opinion
    ( -- business:
      { stage     = 'seed,
        sector    = 'information_technology,
        stage_com = 'pre_profit },
      -- investor:
      { has_ESG = true }
    )

{-
GIVEN   b   IS A Business						
        inv IS A Investor						
GIVETH  opinion IS A Opinion						
DECIDE  opinion IS interesting	IF		stage     IS Seed	
                                  AND   sector    IS Information Technology	
                                  AND   stage_com IS Pre-Revenue	
				                OR      stage     IS Series A	
					              AND   sector    IS Information Technology	
                                  AND   stage_com IS Pre-Profit	
				                OR has_ESG				
	    opinion	IS reject OTHERWISE
-}
