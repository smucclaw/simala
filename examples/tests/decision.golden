get_opinion = <fun/2>
'interesting
